### PR TITLE
10789 rsection basic objects

### PR DESCRIPTION
**Pull request by david-liska**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [ ] Attached examples
- [ ] Own example attached to Pull Request

**Test Configuration**:
* RFEM / RSTAB / RSECTION version:
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: includes/BasicObjects/RsectionElement.js [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+};
+
+/**
+ * Creates RSection arc Element
+ * @param {Number}  no                      Number of Element, can be undefined
+ * @param {Array}   points_of_arc           Points numbers of arc Element
+ * @param {Array}   control_point           Coordinates of control point
+ * @param {Array}   arc_parameters          Arc parameters, can be undefined
+ * @param {Array}   arc_center              Coordinates of arc center, can be undefined
+ * @param {String}  alpha_adjustment_target Subsequent adjustement of alpha by displaycing point at, can be undefined (Begining of arc by default)
```
Comment (jarabroz) [inline includes/BasicObjects/RsectionElement.js:59]: typos adjustment, displaying, beginning

File: includes/BasicObjects/RsectionElement.js [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        this.element.arc_alpha = arc_parameters[2];
+    }
+    if (typeof arc_center !== "undefined") {
+        ASSERT(arc_center.length === 2, "Two parameters must be specified: y, z");
+        this.element.arc_center_y = arc_center[0];
+        this.element.arc_center_z = arc_center[1];
+    }
+    if (typeof alpha_adjustment_target !== "undefined") {
+        if (!(alpha_adjustement_target in alpha_adjustement_target_types)) {
+            console.log("Alpha adjustement tagret " + alpha_adjustement_target + "doesn't exist");
```
Comment (jarabroz) [inline includes/BasicObjects/RsectionElement.js:92]: typo adjustment target

File: includes/BasicObjects/RsectionElement.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+};
+
+/**
+ * Creates RSection circle Element
+ * @param {Number}  no              Number of Element, can be undefined
+ * @param {Array}   circle_center   Coordinates of circle center
+ * @param {Number}  circle_radius   Circle radius
+ * @param {String}  comment         Comment, can be undefined
+ * @param {Object}  params          Parameters, can be undefined
+ * @returns Elementř
```
Comment (jarabroz) [inline includes/BasicObjects/RsectionElement.js:107]: typo Element

File: includes/BasicObjects/RsectionElement.js [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+ */
+RSectionElement.prototype.Parabola = function (no,
+    points_of_parabola,
+    control_point,
+    parabola_alpha,
+    comment,
+    params) {
+    ASSERT(typeof points_of_parabola !== "undefine", "Two points of parabola must be defined");
+    ASSERT(points_of_parabola.length === 2, "Two points must be specified");
+    ASSERT(typeof control_point !== "undefined", "Center point must be deined");
```
Comment (jarabroz) [inline includes/BasicObjects/RsectionElement.js:171]: Denied

File: includes/BasicObjects/RsectionElement.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    ASSERT(control_point.length === 2, "Two parameter must be spocified: y, z");
```
Comment (jarabroz) [inline includes/BasicObjects/RsectionElement.js:172]: specified

File: includes/BasicObjects/RsectionElement.js [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+/**
+ * Shows list of all available alpha adjustement types
+ */
+ function get_alpha_adjustement_types () {
+    console.log(Object.keys(alpha_adjustement_target_types));
+};
+
+const alpha_adjustement_target_types = {
```
Comment (jarabroz) [inline includes/BasicObjects/RsectionElement.js:271]: This could cause problem during compilation on Jenkins

**Review by jarabroz — CHANGES_REQUESTED** [review]

File: examplesRSECTION/high_level_functions_demos/basic_objects_demo.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+/*********************************************************************************************
+****************************************** Main **********************************************
+*********************************************************************************************/
+
+var t1 = new Date().getTime();
+
+run("../includes/Tools/clearAll.js");
+
+var material = createMaterial("S235");
```
Comment (jarabroz) [inline examplesRSECTION/high_level_functions_demos/basic_objects_demo.js:15]: Could be handy to have also some HFL for Materials and Sections

**Review by jarabroz — COMMENTED** [review]
Add also prototypes fopr other types for stress point

File: includes/BasicObjects/RsectionLine.js [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+ * @param {Array}   control_points      Control points ([[y1, z1 (, weight1)], [y2, z2, weight2], ...])
+ * @param {Number}  nurbs_order         Nurbs order, can be undefine (2 as default)
+ * @param {Array}   nurbs_knots         Nurbs knots, can be undefined
+ * @param {String}  comment             Comment, can be undefined
+ * @param {Object}  params              Parameters, can be undefined
+ * @returns Line
+ */
+RSectionLine.prototype.NURBS = function (no,
+    definition_points,
+    control_points,
```
[the remark this answers is not in the document]
Comment (david-liska) [reply includes/BasicObjects/RsectionLine.js:191]: RsectionControlPoint is defined as class (RsectionControlPoint.js), but I can't see no support in RSection API.

File: includes/BasicObjects/RsectionPoint.js [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+ * @class
+ * @constructor
+ * @param {Number}      no              Number of Point, can be undefined
+ * @param {Number}      coordinate_y    Coordinate Y
+ * @param {Number}      coordinate_z    Coordinate Z
+ * @param {String}      comment         Comment, can be undefined
+ * @param {Object}      params          Parameters, can be undefined
+ * @returns Point
+ */
+function RSectionPoint(no,
```
[the remark this answers is not in the document]
Comment (david-liska) [reply includes/BasicObjects/RsectionPoint.js:16]: Podle dohody zatím neřeším. Až bude připravena logika ukládání RSection objektů do jiných adresářů, musí se změnit jak názvy objektů tak upravit ukázkové skripty.

File: examplesRSECTION/high_level_functions_demos/basic_objects_demo.js (hunk 7 above)
Comment (david-liska) [reply examplesRSECTION/high_level_functions_demos/basic_objects_demo.js:15]: Sections are covered, but Materials not (I think there are some missing (un-complete) API support for concrete material values, timber materials values and so on).

**Review by jarabroz — APPROVED** [review]

**Review by jarabroz — APPROVED** [review]

**Pull request merged**